source/features/pr-easy-toggle-files.tsx

```diff
@@ -0,0 +1,48 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const selectorHeaderBar = '.file-header.file-header--expandable';
+
+function toggleFile(event: delegate.Event<Event, HTMLElement>): void {
+	const elementClicked = event.target as HTMLElement;
+	const {parentElement} = elementClicked;
+
+	if (parentElement === null) {
+		return;
+	}
+
+	/**
+	 * A header bar is displayed with flexbox and 2 cells:
+	 *
+	 * 1. Cell on the left
+	 * The largest one, contains the toggle button, the filename...
+	 * Has a large empty area (`.file-header.file-header--expandable > .file-info`) so there is
+	 * a high probability we click on it since it is the easiest.
+	 *
+	 * 2. Cell on the right
+	 * Contains the checkbox "[ ] Viewed" and the 3-dots menu
+	 */
+	if (
+		parentElement.matches(selectorHeaderBar) ||
+		elementClicked.matches(selectorHeaderBar)
+	) {
+		const button = select('button', elementClicked) ?? select('button', parentElement);
```

Comment:
Instead of `delegate`, `.matches` and `select`, you can use this:
```suggestion
	const elementClicked = event.target as HTMLElement;
	const bar = clickedElement.closest(selectorHeaderBar);
	if (bar === clickedElement || bar.parentElement === clickedElement) {
		select(this needs a better selector, bar)!.click();
```

Reply:
I decided to use [`isSameNode()`](https://developer.mozilla.org/en-US/docs/Web/API/Node/isSameNode) instead of `===`.

Reply:
That is the same but longer. No need

Reply:
I wasn't aware of that, good to know thanks!